### PR TITLE
sysext: add a varlink IPC interface

### PR DESCRIPTION
**Pull request by poettering**
Let's continue our journey of adding Varlink interfaces to various components of systemd, in particular those which had no IPC interface so far. In this case sysext

### CODE REVIEW

**Review by bluca — CHANGES_REQUESTED** [review]
Shouldn't this also add a confext service/socket?

**Comment by poettering** [comment]
> Shouldn't this also add a confext service/socket?

You service supports both. You specify in the method call which image class you care about.

File: test/units/testsuite-50.sh [hunk 1]
```diff
@@ -457,6 +457,16 @@ test ! -e /usr/lib/systemd/system/some_file
...
 
+# Similar, but go via varlink
+varlinkctl call /run/systemd/io.systemd.sysext io.systemd.sysext.List '{}'
+(! grep -q -F "MARKER=1" /usr/lib/systemd/system/some_file )
+varlinkctl call /run/systemd/io.systemd.sysext io.systemd.sysext.Merge '{}'
+grep -q -F "MARKER=1" /usr/lib/systemd/system/some_file
+varlinkctl call /run/systemd/io.systemd.sysext io.systemd.sysext.Refresh '{}'
+grep -q -F "MARKER=1" /usr/lib/systemd/system/some_file
+varlinkctl call /run/systemd/io.systemd.sysext io.systemd.sysext.Unmerge '{}'
+(! grep -q -F "MARKER=1" /usr/lib/systemd/system/some_file )
```
Comment (bluca) [inline test/units/testsuite-50.sh:468]: please add confext tests here too, to avoid accidental regressions

**Review by bluca — APPROVED** [review]
Ok, I was going from the angle of having to call `varlinkctl call /run/systemd/io.systemd.sysext` as an interface for confext, which seems a bit confusing, but it can be changed later if it turns out to be, the larger functionality is fine to add immediately

**Pull request merged**